openai_provider.py

```
@@ -0,0 +1,250 @@
+    def __init__(self, api_key: str = None, base_url: str = None):
+        resolved_api_key = api_key or os.environ.get("OPENAI_API_KEY")
+        if not resolved_api_key:
+            raise ValueError("OpenAI API key not provided and not found in OPENAI_API_KEY environment variable. Please set OPENAI_API_KEY or provide the key directly.")
+        try:
+            self.client = openai.OpenAI(api_key=resolved_api_key, base_url=base_url)
+            self.model_identifier = None
+            print("OpenAIProvider initialized successfully.")
+            if base_url:
+                print(f"Using custom OpenAI base URL: {base_url}")
+        except Exception as e:
+            raise ValueError(f"Failed to initialize OpenAI client: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)